File: core/src/main/java/com/canyon/commons/cache/StandardStaticCache.java
package com.canyon.commons.cache;

import java.util.*;

/**
 * 标准静态缓存
 * @param <TKey>
 * @param <TValue>
 */
public class StandardStaticCache<TKey, TValue> implements StaticCache<TKey, TValue> {

	private Map<TKey, TValue> cache = Collections.synchronizedMap(new HashMap<>());

	public int size() {
		return cache.size();
	}

	public List<TValue> all() {
		return new ArrayList<TValue>(cache.values());
	}

	public void add(TKey key, TValue value) {
		cache.put(key, value);
	}

	public void remove(TKey key) {
		cache.remove(key);
	}

	public TValue get(TKey key) {
		return cache.get(key);
	}

	public void clear(){
		cache.clear();
	}
}

File: core/src/main/kotlin/com/canyon/commons/cache/StandardStaticCache.kt
package com.canyon.commons.cache

import java.util.concurrent.ConcurrentHashMap

class StandardStaticCache<TKey, TValue> : StaticCache<TKey, TValue> {

    private val cache = ConcurrentHashMap<TKey, TValue>()

    override fun size(): Int {
        return cache.size
    }

    override fun all(): List<TValue> {
        return cache.elements().toList()
    }

    override fun add(key: TKey, value: TValue) {
        cache[key] = value
    }

    override fun remove(key: TKey) {
        cache.remove(key)
    }

    override fun get(key: TKey): TValue? {
        return cache.get(key)
    }

    override fun clear() {
        cache.clear()
    }

}